### DISCUSSION TITLE
How to manually load zsh completion definition?

### DISCUSSION THREAD

**Discussion by lazywei** [marlonrichert/zsh-snap#238] [Q&A] [answered]
I want to load the completion from a non-canonical repo:
https://github.com/facebook/sapling/blob/main/eden/scm/contrib/zsh_completion

Is there a way to manually load a file that contains the completion definitions? I tried to add this to `$HOME/.local/share/zsh/site-functions` but it doesn't seem to work.

**Comment by marlonrichert** [comment, marked as answer]
Add the following to your `.zshrc` file:
```sh
znap clone facebook/sapling
znap fpath _hg '< ~[facebook/sapling]/eden/scm/contrib/zsh_completion'
```

**Reply by lazywei** [reply]
Got it, thank you!

**Comment by ppwwyyxx** [comment]
Can I request a mode/option of zsh-snap to let it not do anything about zsh's existing comp system?

It seems to do some magic which I don't understand and it seems the "normal" way of using fpath/compdef/compdump does not work when using zsh-snap. This is a concern for me because other people's completion code (by calling compdef) written for "normal" zsh does not work for me.

**Reply by marlonrichert** [reply]
It should work with other people's completion code. At least it works for me. Can you give me a reproducible test case?
